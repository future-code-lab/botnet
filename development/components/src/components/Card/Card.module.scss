.card {
  display: flex;
  flex-wrap: wrap;
  min-width: 450px;
  justify-content: center;
  width: 450px;
  border-radius: 8px;
  background: #ffffff;
  border: 2px solid rgba(249, 250, 252, 0.5);
  box-sizing: border-box;
  box-shadow: 0px 4px 145px rgba(0, 0, 0, 0.05);
}

.header {
  font-size: 30px;
  color: #000000;
  line-height: 37px;
  width: 100%;
  height: min-content;
  justify-content: flex-start;
  display: flex;
  font-family: 'Montserrat', sans-serif;
  font-style: normal;
  font-weight: 500;
  padding: 0 20px;
}

.content {
  width: 100%;
  color: #858a8f;
  font-family: 'Inter', sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 25px;
  line-height: 30px;
  padding: 0px 20px 20px 20px;
}

.image {
  width: 100%;
  height: 200px;
}

.button {
  min-width: 160px;
  text-align: left;
}

.buttonContainer {
  width: 100%;
  display: flex;
  justify-content: flex-start;
  padding: 20px;
}
