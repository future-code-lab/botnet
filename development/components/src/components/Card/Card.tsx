import React, { ReactNode } from 'react';

import cx from 'classnames';

import { PrimaryButton } from '../Button/PrimaryButton';
import { Header } from '../Header';
import { HeaderSizes } from '../Header/Header.types';

import css from './Card.module.scss';

export interface CardProps {
  className?: string;
  buttonText?: string;
  onButtonClick?: () => void;
  header?: string;
  withImage?: boolean;
  withHeader?: boolean;
  withChildren?: boolean;
  withButton?: boolean;
  imageSrc?: string;
  imageAlt?: string;
  children?: ReactNode;
}

export const Card: React.FC<CardProps> = ({
  buttonText,
  onButtonClick,
  className,
  header,
  imageSrc,
  imageAlt,
  children,
  withHeader,
  withChildren,
  withButton,
  withImage,
}) => {
  return (
    <div className={cx(css.card, className)}>
      {withImage && <img className={css.image} src={imageSrc} alt={imageAlt} />}
      {withHeader && (
        <Header className={css.header} size={HeaderSizes.medium}>
          {header}
        </Header>
      )}
      {withChildren && <div className={css.content}>{children}</div>}
      {withButton && (
        <div className={css.buttonContainer}>
          <PrimaryButton className={css.button} text={buttonText} onClick={onButtonClick} />
        </div>
      )}
    </div>
  );
};
