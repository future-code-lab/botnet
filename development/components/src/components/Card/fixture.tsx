import React from 'react';
import { CardProps } from './Card';

export const propsFixture: CardProps = {
  className: 'className',
  buttonText: 'button text',
  onButtonClick: () => {},
  withButton: true,
  header: 'header',
  withHeader: true,
  withChildren: true,
  withImage: true,
  imageSrc: 'background.png',
  imageAlt: 'alt',
  children: (
    <div style={{ width: '100%', display: 'flex', flexWrap: 'wrap', justifyContent: 'flex-start' }}>
      <span style={{ width: '100%' }}>Ехал Грека через реку. </span>
      <span style={{ width: '100%' }}> Видит Грека в реке рак. </span>
      <span style={{ width: '100%' }}>Сунул в реку руку Грека.</span>
      <span style={{ width: '100%' }}> Рак за руку Греку - цап. </span>
    </div>
  ),
};

export const propsFixtureRequiredFields: CardProps = {};

export const propsFixtureWithImageOnly: CardProps = {
  withImage: true,
  imageSrc: 'image src',
  imageAlt: 'alt',
};

export const propsFixtureWithButtonOnly: CardProps = {
  buttonText: 'button text',
  onButtonClick: () => {},
  withButton: true,
};

export const propsFixtureWithHeaderOnly: CardProps = {
  header: 'header',
  withHeader: true,
};

export const propsFixtureWithChildrenOnly: CardProps = {
  children: 'children',
  withChildren: true,
};
