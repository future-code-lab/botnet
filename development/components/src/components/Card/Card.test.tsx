import React from 'react';
import { render } from '@testing-library/react';
import { Card } from './Card';
import {
  propsFixture,
  propsFixtureRequiredFields,
  propsFixtureWithImageOnly,
  propsFixtureWithButtonOnly,
  propsFixtureWithHeaderOnly,
  propsFixtureWithChildrenOnly,
} from './fixture';

test('renders Card component', () => {
  const component = render(<Card {...propsFixture} />);
  expect(component).toMatchSnapshot();
});

test('renders Card component with required fields children', () => {
  const component = render(<Card {...propsFixtureRequiredFields} />);
  expect(component).toMatchSnapshot();
});

test('renders Card component with image only', () => {
  const component = render(<Card {...propsFixtureWithImageOnly} />);
  expect(component).toMatchSnapshot();
});

test('renders Card component with button only', () => {
  const component = render(<Card {...propsFixtureWithButtonOnly} />);
  expect(component).toMatchSnapshot();
});

test('renders Card component with header only', () => {
  const component = render(<Card {...propsFixtureWithHeaderOnly} />);
  expect(component).toMatchSnapshot();
});

test('renders Card component with children only', () => {
  const component = render(<Card {...propsFixtureWithChildrenOnly} />);
  expect(component).toMatchSnapshot();
});
