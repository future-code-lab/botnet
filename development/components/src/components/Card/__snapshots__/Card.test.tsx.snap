// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card className"
      >
        <img
          alt="alt"
          class="image"
          src="background.png"
        />
        <div
          class="header"
        >
          <h3>
            header
          </h3>
        </div>
        <div
          class="content"
        >
          <div
            style="width: 100%; display: flex; flex-wrap: wrap; justify-content: flex-start;"
          >
            <span
              style="width: 100%;"
            >
              Ехал Грека через реку. 
            </span>
            <span
              style="width: 100%;"
            >
               Видит Грека в реке рак. 
            </span>
            <span
              style="width: 100%;"
            >
              Сунул в реку руку Грека.
            </span>
            <span
              style="width: 100%;"
            >
               Рак за руку Греку - цап. 
            </span>
          </div>
        </div>
        <div
          class="buttonContainer"
        >
          <button
            class="button text primary button"
          >
            button text
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card className"
    >
      <img
        alt="alt"
        class="image"
        src="background.png"
      />
      <div
        class="header"
      >
        <h3>
          header
        </h3>
      </div>
      <div
        class="content"
      >
        <div
          style="width: 100%; display: flex; flex-wrap: wrap; justify-content: flex-start;"
        >
          <span
            style="width: 100%;"
          >
            Ехал Грека через реку. 
          </span>
          <span
            style="width: 100%;"
          >
             Видит Грека в реке рак. 
          </span>
          <span
            style="width: 100%;"
          >
            Сунул в реку руку Грека.
          </span>
          <span
            style="width: 100%;"
          >
             Рак за руку Греку - цап. 
          </span>
        </div>
      </div>
      <div
        class="buttonContainer"
      >
        <button
          class="button text primary button"
        >
          button text
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders Card component with button only 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <div
          class="buttonContainer"
        >
          <button
            class="button text primary button"
          >
            button text
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <div
        class="buttonContainer"
      >
        <button
          class="button text primary button"
        >
          button text
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders Card component with children only 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <div
          class="content"
        >
          children
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <div
        class="content"
      >
        children
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders Card component with header only 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <div
          class="header"
        >
          <h3>
            header
          </h3>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <div
        class="header"
      >
        <h3>
          header
        </h3>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders Card component with image only 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <img
          alt="alt"
          class="image"
          src="image src"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <img
        alt="alt"
        class="image"
        src="image src"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders Card component with required fields children 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
